frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -840,68 +840,94 @@ function ReaderPaging:onGotoPageRel(diff)
     logger.dbg("goto relative page:", diff)
     local new_va = self.visible_area:copy()
     local x_pan_off, y_pan_off = 0, 0
+    local right_to_left = (self.ui.document.configurable.writing_direction == 1)
 
     if self.zoom_mode:find("width") then
         y_pan_off = self.visible_area.h * diff
     elseif self.zoom_mode:find("height") then
         -- negative x panning if writing direction is right to left
-        local direction = self.ui.document.configurable.writing_direction
-        x_pan_off = self.visible_area.w * diff * (direction == 1 and -1 or 1)
-    elseif self.zoom_mode:find("column") then
-        -- zoom mode for two-column navigation
-
-        y_pan_off = self.visible_area.h * diff
-        y_pan_off = Math.roundAwayFromZero(y_pan_off)
-        new_va.x = Math.roundAwayFromZero(self.visible_area.x)
-        new_va.y = Math.roundAwayFromZero(self.visible_area.y+y_pan_off)
-        -- intra-column navigation (vertical), this is the default behavior
-        -- if we do not reach the end of a column
+        x_pan_off = self.visible_area.w * diff * (right_to_left and -1 or 1)
+
+    elseif self.zoom_mode == "pan" or self.zoom_mode == "column" then
+        -- pan zoom mode
+
+        local bottom_to_top = self.ui.doc_settings:readSetting("bottom_to_top")
```

Comment:
Looks like you never create/save a setting named "bottom_to_top", so typo? Did you mean `zoom_pan_bottom_to_top`?
(I was going to suggest this is a bad name as not explicite enough when one would skim thru a setting file. `zoom_pan_bottom_to_top` is fine.)

Reply:
Sorry; should I have tested more extensively, I'd seen this one…